discovery-infra/tests/base_test.py

```diff
@@ -138,6 +138,26 @@ def attach(node, disk_size, bootable=False):
         for modified_node in modified_nodes:
             modified_node.detach_all_test_disks()
 
+    @pytest.fixture()
+    def attach_interface(self):
+        added_networks = []
+
+        def add(node, network_name=None, network_xml=None):
+            if network_xml:
+                network, interface_mac = node.attach_interface(network_xml)
+            elif network_name:
+                interface_mac = node.add_interface(network_name)
+                network = node.get_network_by_name(network_name)
+            added_networks.append({"node": node, "network": network, "mac": interface_mac})
+
+        yield add
+        for added_network in added_networks:
+            logging.info(f'Deleting custom networks:{added_networks}')
+            with suppress(Exception):
```

Comment:
Please use specific exceptions.

Reply:
but there could be few

Reply:
So address all of them

Reply:
In this case we need it to be general Exception, as undefine_interface uses run_command while destroy_network uses libvirt object